### PR TITLE
Support Omnipay v3, as Omnipay v2 hasn't seen active development since 2016

### LINKED ISSUE #196: Omnipay V3 support
I was wondering, has anyone looked into omnipay 3.x yet? Is there a lot of difference?

What will need doing to upgrade this module to 3.x support?

### PR DESCRIPTION
**Pull request by hchokshi**
There is minimal user exposed change in the API. Majority of change is in how gateways internally handle sending requests (via `ClientInterface`, rather than an explicit dependency on Guzzle).

Unfortunately, there is currently a failing test due to a stray reference to Guzzle in the `omnipay/paymentexpress` gateway package. All tests pass once this is resolved - thephpleague/omnipay-paymentexpress#50, thephpleague/omnipay-paymentexpress#51

Resolves #196

### CODE REVIEW

**Comment by bummzack** [comment]
Thanks for this. Maybe we could also refactor tests, so that they don't use the paymentexpress gateway at all?

**Comment by hchokshi** [comment]
Hey @bummzack there are quite a few tests that are dependent on the `PaymentExpress_PxPay` gateway, and I don't feel confident in my familiarity with the codebase in maintaining adequate coverage switching it for a different gateway. 

The PR thephpleague/omnipay-paymentexpress#51 has been merged, and I've requested the maintainer issue a patch release. Once that's done, it should be as easy as just requesting a rebuild on Travis.

**Comment by bummzack** [comment]
Please don't add other functionality to this PR. This PR should only contain the Omnipay 3 update. For the other features, create new PRs. Thanks a lot ❤️

**Comment by hchokshi** [comment]
Note: commit `Make transaction ID length configurable` was made before comment https://github.com/silverstripe/silverstripe-omnipay/pull/201#issuecomment-418618980

I sent an empty amend to re-trigger the travis build, as `omnipay/paymentexpress` has tagged a release `3.0.1` with the PR fixes mentioned earlier.

I'm now working to resolve the build errors.

File: src/Model/Payment.php [hunk 1]
```diff
@@ -81,6 +83,13 @@
 
     private static $default_sort = '"Created" DESC, "ID" DESC';
 
+    /**
+     * Length to use for newly generated transaction identifiers.
+     * @config
+     * @var int
+     */
+    private static $transaction_id_length = 30;
```
Comment (bummzack) [inline src/Model/Payment.php]: Please remove this feature from the PR.

File: src/Model/Payment.php [hunk 2]
```diff
@@ -559,9 +570,11 @@ protected function generateUniquePaymentIdentifier()
     {
         $generator = Injector::inst()->create(RandomGenerator::class);
         $id = null;
+        $length = static::config()->get('transaction_id_length');
```
Comment (bummzack) [inline src/Model/Payment.php]: Same here. I think it's a good idea, but should not be part of this PR. Also, maybe a token configuration per gateway could be beneficial? Who knows what silly requirements other gateways have.

**Comment by hchokshi** [comment]
I've dropped the transaction ID commit from this PR.

**Comment by hchokshi** [comment]
Hey @bummzack, can you please tag a release with these changes? Thanks.
